src/sentry.js

```diff
@@ -1,15 +1,75 @@
 /* @flow strict-local */
 import * as Sentry from '@sentry/react-native';
+import type { Options } from '@sentry/react-native';
 
 import config from './config';
 
-if (config.enableSentry) {
-  Sentry.init({
-    dsn: config.sentryKey,
-    deactivateStacktraceMerging: true,
-    ignoreErrors: ['Network request failed'],
-  });
-} else {
+/** Initialize Sentry, possibly into a disabled state. */
+//
+// There appears to be exactly one line of official documentation about
+// initializing Sentry into a disabled state:
+//
+// https://docs.sentry.io/development/sdk-dev/overview/#usage-for-end-users
+// > If an empty DSN is passed, you should treat it as valid option which
+// > signifies disabling the SDK.
+//
+// Unfortunately, this is not actually documentation for our SDK; it's advice to
+// developers of individual languages' Sentry SDKs about what end-users should
+// see. In particular:
+//  * The TypeScript type declarations for the JavaScript family of SDKs [1]
+//    define the the Options struct as having `dsn?: string`, disallowing
+//    `null`.
+//  * If you ignore that and pass in `null` anyway, the Sentry Java SDK (used
+//    under the hood on Android) treats a null or empty DSN as worthy of
+//    warning.
+//
+// Fortunately, there is also an almost-undocumented `enabled?: boolean` option
+// [2] available to us. The combination of an invalid DSN and `enabled: false`
+// is not documented, but has not been observed to have ill effects.
+//
+// (A note for future work:
+//
+//   A comment by a developer [3] indicates that `enabled` can even be altered
+//   at runtime, but this is not to be done blindly. As described, that _will_
+//   disable Sentry; but initializing it in a disabled state and enabling it
+//   afterwards will leave Sentry's integrations uninitialized. [4]
+//
+//   A better method may be to call `init()` again, even though the "Static API"
+//   documentation [5] says this is undefined. A still-better method might be to
+//   disregard the "Static API" and use `getCurrentHub()` everywhere.)
+
+// [1] https://github.com/getsentry/sentry-javascript/blob/7a792f4e/packages/types/src/options.ts#L25
+// [2] https://github.com/getsentry/sentry-javascript/blob/7a792f4e/packages/types/src/options.ts#L19
+// [3] https://github.com/getsentry/sentry-javascript/issues/2039#issuecomment-486674574
+// [4] https://github.com/getsentry/sentry-javascript/blob/7a792f4e/packages/core/src/baseclient.ts#L76
+// [5] https://docs.sentry.io/development/sdk-dev/unified-api/#static-api
```

Comment:
Can we just never call Sentry when we have it disabled? It seems like it wouldn't be too hard to have our own boolean defined in this file to track that; and then in our handful of direct call sites to Sentry, we could just check that boolean first. Those are already concentrated into a handful of wrappers, for good reasons independent of this aspect.

If that'd let us not have to worry about any of the complexity described in this comment, that seems like it'd be an excellent choice.

Reply:
We could certainly check isEnabled at call sites, but we'll still have to worry about this eventually: at some point we'll need to be able to completely enable or disable Sentry at runtime, not just startup time, and then all this becomes relevant again.

(In particular, merely toggling that hypothetical boolean wouldn't suffice, as it wouldn't disable the unhandled-exception handlers.)

Reply:
> at some point we'll need to be able to completely enable or disable Sentry at runtime, not just startup time

Will we? I'm curious about the use cases you have in mind; and will be interested in thinking about what solutions we can come up with for them.

Possibly a chat thread would be a good place for that discussion.